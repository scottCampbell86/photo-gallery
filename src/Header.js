import React, { Component } from 'react';

export default class Header extends Component {
  render() {
    return (
      <>
        <header className="App-header">
          <img src="" alt=""/>
          <h1>Monster Cards</h1>
        </header>  
      </>
    );
  }
}