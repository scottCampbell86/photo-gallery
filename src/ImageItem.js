import React, { Component } from 'react'


export default class ImageItem extends Component {
    render() {
        return (
            <div className="image-container">
                <li>
                    <h2>{this.props.image.title}</h2>
                    <p>{this.props.image.keyword}</p>
            
                    <img
                        alt={this.props.image.url}
                        src={this.props.image.url} />
                </li>
            </div>

            
        )    
    }
}