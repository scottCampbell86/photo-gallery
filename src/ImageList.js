import React, { Component } from 'react'

export default class images extends Component {
    render() {
        return <li>
            <div className="image-container">
                <h2>{this.props.feline.name}</h2>
                <p className="cat-type">{this.props.feline.type}</p>
            </div>

            <div className="image-container">
                <img
                    alt={this.props.feline.url}
                    src={this.props.feline.url} />
            </div>
            <p className="year">{this.props.feline.year}</p>
        </li>
    }
}