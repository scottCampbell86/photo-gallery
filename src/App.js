import React, { Component } from 'react';
import './App.css';
import images from './data.js';
import Header from './Header.js';
import ImageList from './ImageList.js';
import DropDown from './DropDown.js';

export default class App extends Component {
  state = { selected: null };

    render() {
    {console.log(images);}
    const selected = this.state.selected;     
    const handleChange = e => {
        this.setState({ selected: e.target.value });
    };
    const filteredArray = images.filter(image => {
      if (!selected) return true;
      return image.keyword === selected;
    })
    {console.log(filteredArray);}
   
    return (
      <>
        <Header />
        <main>
          <section className="options"> 
              <DropDown handleChangeCallback={handleChange}/>
          </section>
          
          <section>
              <ImageList imagesArray={filteredArray} />
          </section>
        </main>
      </>       
      );
    };
  };