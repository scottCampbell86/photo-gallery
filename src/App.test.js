import React from 'react';
//import { render } from '@testing-library/react';
import App from './App.js';
import Header from './Header.js';
import ImageItem from './ImageItem.js';
import ImageList from './ImageList.js';


//import Link from '../Link.react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Header />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<ImageList />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<ImageItem />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
