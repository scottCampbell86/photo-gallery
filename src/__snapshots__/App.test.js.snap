// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header>
  <img
    alt=""
    src=""
  />
  <h1>
    Horned Beasts
  </h1>
</header>
`;
